client/src/app.tsx

```diff
@@ -130,7 +132,10 @@ export function App(appProps) {
         <WritersHomepage />
       </Layout>
     ) : (
-      <PageOrPageNotFound pageNotFound={appProps.pageNotFound}>
+      <PageOrPageNotFound
+        pageNotFound={appProps.pageNotFound}
+        initialPathname={initialPathname}
```

Comment:
Could we make the decision here by introducing a `pageNotFound` state that we set to false/true in a `useEffect()` that compares initialPathname and pathname and uses `appProps.pageNotFound` only if they're equal? This should allow us to eliminate the logic inside `PageOrPageNotFound`.

Reply:
Went ahead and did the refactoring in 44086ba9f430ffa250e1097a6e4f5a477169d4c7 to land this.